cymetric/convenient_interface.py

```diff
@@ -0,0 +1,584 @@
+    df.reset_index(inplace=True)
+
+    time = evaler_.eval('TimeList')
+    df = add_missing_time_step(df, time)
+    return df
+
+
+def get_transaction_activity_timeseries(evaler_, send_list=[], rec_list=[], commod_list=[], nuc_list=[]):
+    """
+    Shape the reduced transation Data Frame into a simple time serie. Applying nuclides selection when required.
+
+    Parameters
+    ----------
+    evaler_ : evaler
+    send_list : list of the sending facility
+    rec_list : list of the receiving facility
+    commod_list : list of the receiving facility
+    nuc_list : list of nuclide to select.
+    """
+
+    if len(nuc_list) != 0:
+        nuc_list = format_nuclist(nuc_list)
+
+    df = get_transaction_activity_df(evaler_, send_list, rec_list, commod_list,
+                                     nuc_list)
+
+    group_end = ['ReceiverProto', 'SenderProto', 'Time']
+    group_start = group_end + ['Activity']
+    df = df[group_start].groupby(group_end).sum()
+    df.reset_index(inplace=True)
+
+    df = df[['Time', 'Activity']].groupby(['Time']).sum()
+    df.reset_index(inplace=True)
+
+    time = evaler_.eval('TimeList')
+    df = add_missing_time_step(df, time)
+    return df
+
+
+def get_transaction_decayheat_timeseries(evaler_, send_list=[], rec_list=[], commod_list=[], nuc_list=[]):
+    """
+    Shape the reduced transation Data Frame into a simple time serie. Applying nuclides selection when required.
+
+    Parameters
+    ----------
+    evaler_ : evaler
+    send_list : list of the sending facility
+    rec_list : list of the receiving facility
+    commod_list : list of the receiving facility
+    nuc_list : list of nuclide to select.
+    """
+
+    if len(nuc_list) != 0:
+        nuc_list = format_nuclist(nuc_list)
+
+    df = get_transaction_decayheat_df(evaler_, send_list, rec_list, commod_list,
+                                      nuc_list)
+
+    group_end = ['ReceiverProto', 'SenderProto', 'Time']
+    group_start = group_end + ['DecayHeat']
+    df = df[group_start].groupby(group_end).sum()
+    df.reset_index(inplace=True)
+
+    df = df[['Time', 'DecayHeat']].groupby(['Time']).sum()
+    df.reset_index(inplace=True)
+
+    time = evaler_.eval('TimeList')
+    df = add_missing_time_step(df, time)
+    return df
+
+
+def get_inventory_df(evaler_, fac_list=[], nuc_list=[]):
+    """
+    Shape the reduced inventory Data Frame. Applying nuclides/facilities selection when required.
+
+    Parameters
+    ----------
+    evaler_ : evaler
+    fac_name : name of the facility
+    nuc_list : list of nuclide to select.
+    """
+
+    # Get inventory table
+    df = evaler_.eval('ExplicitInventory')
+    agents = evaler_.eval('AgentEntry')
+
+    rdc_list = []  # because we want to get reed of the nuclide asap
+    if len(nuc_list) != 0:
+        nuc_list = format_nuclist(nuc_list)
+        rdc_list.append(['NucId', nuc_list])
+    else:
+        wng_msg = "no nuclide provided"
```

Comment:
since nuc_list gets passed through several functions before reaching this one, perhaps the warning message should come the very first time it is checked?